src/services/convert.service.ts

```diff
@@ -0,0 +1,51 @@
+import { convert } from '@asyncapi/converter';
+import { AsyncAPIDocument } from '@asyncapi/parser';
+import specs from '@asyncapi/specs';
+
+import YAML from 'js-yaml';
+
+import { logger } from '../utils/logger';
+
+/**
+ * Service providing `@asyncapi/converter` functionality.
+ */
+export class ConvertService {
+  /**
+   * Convert the given spec to the desired language.
+   * @param spec AsyncAPI spec
+   * @param language Language to convert to, YAML or JSON
+   * @param version [version] AsyncAPI spec version
+   * @returns converted spec
+   */
+  public async convertSpec(
+    spec: AsyncAPIDocument | string,
+    language = '',
+    version: string = this.getLastVersion(),
+  ): Promise<string> {
+    try {
+      const asyncapiSpec = typeof spec === 'object' ? JSON.stringify(spec) : spec;
+      const convertedSpec = convert(asyncapiSpec, version);
+
+      return language === 'json'
+        ? this.convertToJSON(convertedSpec)
+        : convertedSpec;
+    } catch (err) {
+      logger.error('[ConvertService] An error has occurred while converting spec to version: {0}. Error: {1}', version, err);
```

Comment:
sorry I didn't understand your suggestion, should we make a new Error (defined on the exceptions folder) and throw that instead of `err`? Or is your comment on the fact that we log the error?

Reply:
hi @magicmatatjahu, so could we create a `ConvertExcetion` and throw a new error of that type. I think the log adds a little bit of value in knowing where the error happened, but we can also remove the `catch` if you want, wdyt?

Reply:
Hmm 🤔  I can create PR in `converter-js` to throw errors here https://github.com/asyncapi/converter-js/blob/master/lib/index.js#L37 and we can go with solution like in other controllers, with returning `ProblemException` and with proper type like `.../problems/converter-...`. What do you think?

Reply:
Sorry for that late response! Really sorry, I have a hundreads of notifications 😅

Reply:
> we can go with solution like in other controllers, with returning ProblemException and with proper type like .../problems/converter-.... What do you think?

I think we can do that yes, but I'm just a little hesitant on doing that in case these `service` classes are turned into a reusable package ([discussed here](https://github.com/asyncapi/community/discussions/212)). Mainly because only this HTTP API knows about the media type `application/problem+json`, the CLI wouldn't.
But for now let's use `ProblemException` 👍

Reply:
> I think we can do that yes, but I'm just a little hesitant on doing that in case these service classes are turned into a reusable package (https://github.com/asyncapi/community/discussions/212). Mainly because only this HTTP API knows about the media type application/problem+json, the CLI wouldn't.
But for now let's use ProblemException 👍

Ok, I understand :) I think the whole idea in the attached discussion makes sense, but instead of a separate package we should write such services in source code of libraries, but the topic needs to be discussed.

> I added the ProblemException but we should review the type after the other PR, and make some tests on the converter-js or on this convert.service

Ok, I will create PR with exceptions in the `converter-js` :) However `ProblemExceptions` should only be created in this package. The `converter-js` will only throw an error without any metadata like title, detail, path etc.

Reply:
> and we can go with solution like in other controllers, with returning ProblemException and with proper type like .../problems/converter-.... What do you think?

@magicmatatjahu should we change the `type` field to point towards the README of the `converter` repo, and add something like this: https://github.com/asyncapi/parser-js#error-types?
Currently we would point to `https://api.asyncapi.com/problem/converter` or `https://api.asyncapi.com/problem/converter-json`, but we don't handle those routes on the API... should we? I think it might be better to handle this on GitHub rather than render HTML or markdown in this API for Problem details, wdyt?

Reply:
@BOLT04 Yes, you're right. Atm we don't handle such an endpoint (I mean `problem/...`) in our API, but we should. I created few days ago GSOC proposal to handle errors in our all packages using that `problem` interface - https://github.com/asyncapi/community/issues/266 One of the action point (but nice to have, not mandatory) is integration registry of all problem types in our organization. Pointing to the `Readme` of project is nice, but better option will be point to `api.server.com`. I know that errors throw by parser-js point to the Readme, but we should consolidate it.

Having api which will return details of problem we can then easy integrate with Studio and render info about problemo in nice way :) So I also see benefits to have one place where we will have described problems.

In the `problem` interface we can also add additional fields (other than `type`, `details` etc) so we can add `project` field with `url` which points to the project:

```ts
{
  url: 'https://github.com/asyncapi/converter',
  problemUrl: ...
}
```

What do you think? Do you need some help with your code? I could help you, because I'm remorseful 😅

Reply:
Also I think that we should describe all problems throw by package in the Readme as we do in the `parser-js` :)

Reply:
> Also I think that we should describe all problems throw by package in the Readme as we do in the `parser-js` :)

completely agreed 👍 

> Having api which will return details of problem we can then easy integrate with Studio and render info about problemo in nice way :) So I also see benefits to have one place where we will have described problems.

Yes that would be awesome! Then having more control about the result of the endpoint inside `type` is better, instead of just pointing to a section in a `README`. But these endpoints will be on this API and that `@asyncapi/problem` package will handle all the logic for the endpoint? Could you specify this part a bit more 🙂 ?

Reply:
@BOLT04 ❤️ 

The `@asyncapi/problem` package itself should be the library with which we should create the problem instance itself. It should not have any logic related to ServerAPI. The implementation of this endpoint itself should be here. I don't know how registry of problems should look like so that ServerAPI will be up-to-date with all the problems we have in the organization. This is a topic for discussion :)